dashboard/app/controllers/certificate_images_controller.rb

```diff
@@ -2,9 +2,11 @@
 
 class CertificateImagesController < ApplicationController
   # GET /certificate_images/filename.jpg
+  # filename includes three encoded params:
+  #   name - student name (required)
+  #   course - course name (optional)
+  #   sponsor - donor name (required)
```

Comment:
note that name and sponsor are required here, but optional in other controllers. I chose this because (1) we need to require sponsor here in order for this page to be cacheable, but (2) we need to be able to render the /certificates and /print_certificates pages without a name or sponsor (we use static images instead of /certificate_images in that case).

Reply:
great call Maureen, there was not a good reason to use both donor and sponsor. I've changed the new param to donor.